evap/evaluation/models_logging.py

```diff
@@ -358,18 +358,25 @@ def _m2m_changed(sender, instance, action, reverse, model, pk_set, **kwargs):  #
     if not issubclass(model_class, LoggedModel):
         return
 
-    if reverse:
-        match action:
-            case "pre_remove":
-                action_type = FieldActionType.M2M_REMOVE
-            case "pre_add":
-                action_type = FieldActionType.M2M_ADD
-            case "pre_clear":
-                # Since we are not clearing the LoggedModdel instance, we need to log the removal of the related instances
-                action_type = FieldActionType.M2M_REMOVE
-            case _:
-                return
+    match action:
+        case "pre_remove":
+            action_type = FieldActionType.M2M_REMOVE
+        case "pre_add":
+            action_type = FieldActionType.M2M_ADD
+        case "pre_clear":
+            action_type = FieldActionType.M2M_CLEAR
```

Comment:
Can you explain why the changes here are necessary, and how this is now supposed to work for the reverse-clear case? My understanding is that if we do a reverse-clear (i.e., `A.b_instances.clear()`), we want to treat it as N forward removes (i.e., as if `B.a_instances.remove()` was called on each B that was in `A.b_instances`). This, in my head, should still work perfectly fine with a generic "If adding or removing an empty set, bail out early" logic below -- and it should even require it: If `A.b_instances` was already empty, we don't want to trigger logging, but right now, your early-out wouldn't catch this case.